python/arcpy-python/count_multipart_features_2/count_multipart_features_2.py

```diff
@@ -0,0 +1,148 @@
+                    report_only=False):
+    """This function will take an input feature class and add a new field
+    featureclass: The path to the feature class
+    field_name: The name of the count field to be added ("PartCount" by default)
+    """
+    # Get the Feature object
+    features = Feature(fc_path)
+    
+    # Set the workspace to the feature class workspace
+    arcpy.env.workspace = features.workspace_path
+    
+    # MultiPatch is not supported for multipart features
+    if features.shape_type == "MultiPatch":
+        raise ValueError("This is not a supported geometry shape type. Please select a Multipoint, Polyline, or Polygon")
+    
+    # Count the number of parts for each multipart feature and add it to a dictionary
+    multipart_counts = \
+        {
+            row[features.id_field]: row[features.shape_field].partCount  # Get the number of parts for each multipart
+            for row in features.get_rows([features.id_field, features.shape_field])
+            if row[features.shape_field].isMultipart  # Only get the rows that are multipart
+        }
+    
+    # Don't bother updating the rows if there are no multipart features
+    if len(multipart_counts) == 0:
+        print("No multipart features found")
+        return
+    
+    if not report_only:  # Only set up the output field and update the rows if we are not just reporting
+        # Set up the output field
+        if field_name in features.field_names:
+            if not overwrite:
+                raise ValueError(f"The field {field_name} already exists in {fc_path}")
+            else:
+                arcpy.DeleteField_management(fc_path, field_name)
+
+        arcpy.AddField_management(fc_path, field_name, 'SHORT')  # Add the field
+    
+        # Update the rows with part counts
+        with arcpy.da.Editor(features.workspace_path):
+            upd_keys = [str(k) for k in multipart_counts.keys()]
+            # Use the OIDFieldName to build the SQL query, OBJECTID and OID@ dont work in queries
+            update_query = f"{features.OIDFieldName} IN ({','.join(upd_keys)})"  # Only update the rows that are in the dictionary
+            # Use _update_rows to get a row dictionary so we can update the row using field names
+            for cursor, row in features.update_rows([features.id_field, field_name], query=update_query):
+                row[field_name] = multipart_counts[row[features.id_field]]  # Get the part count from the dictionary
+                cursor.updateRow(list(row.values()))  # Convert the dictionary to a list and update the row
+                     
+    print(f"{len(multipart_counts)} multipart features found in {features.baseName}")
+
+def main():
+    
+    # Set these
+    feature_class = r'path\to\your\feature_class'
+    count_field = 'PartCount'
+    overwrite = True
+    report_only = False
+    
+    count_multipart(feature_class, count_field=count_field, overwrite=overwrite, report_only=report_only)
```

Comment:
This functioncall was malformed on the last commit, `count_field=` needs to be `field_name=`